src/compiler/checker.ts

```diff
@@ -9746,9 +9915,22 @@ namespace ts {
                         }
                     }
                 }
+                else if (source.flags & TypeFlags.Conditional) {
+                    const constraint = getConstraintOfDistributiveConditionalType(<ConditionalType>source);

```

Comment:
Does this constraint here need to be instantiated with a this type of the distributive conditional?